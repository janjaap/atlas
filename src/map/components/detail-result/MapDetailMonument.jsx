import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from './MapDetailResultItem';
import MapDetailResultWrapper from './MapDetailResultWrapper';

const MapDetailMonument = ({ panoUrl, monument, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={monument.label}
    title="Monument"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Nummer"
        value={`${monument.number}`}
      />
      <MapDetailResultItem
        label="Type"
        value={monument.type}
      />
      <MapDetailResultItem
        label="Status"
        value={monument.status}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailMonument.propTypes = {
  monument: PropTypes.shape({
    label: PropTypes.string,
    number: PropTypes.number,
    status: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailMonument;
