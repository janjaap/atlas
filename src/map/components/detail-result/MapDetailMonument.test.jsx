import React from 'react';
import { shallow } from 'enzyme';

import MapDetailMonument from './MapDetailMonument';

describe('MapDetailMonument', () => {
  it('should render everything', () => {
    const monument = {
      label: 'Monument label',
      number: 123456,
      status: 'Monument status',
      type: 'Monument type'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailMonument
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        monument={monument}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
