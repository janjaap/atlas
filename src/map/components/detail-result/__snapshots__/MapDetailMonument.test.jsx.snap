// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailMonument should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Monument label"
  title="Monument"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Nummer"
      value="123456"
    />
    <MapDetailResultItem
      label="Type"
      value="Monument type"
    />
    <MapDetailResultItem
      label="Status"
      value="Monument status"
    />
  </ul>
</MapDetailResultWrapper>
`;
